src/components/app/data/hooks/useEnterpriseCustomer.test.jsx

```diff
@@ -0,0 +1,95 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { AppContext } from '@edx/frontend-platform/react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { authenticatedUserFactory } from '../services/data/__factories__';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+import { queryClient } from '../../../../utils/tests';
+import { fetchEnterpriseLearnerData } from '../services';
+import { useEnterpriseLearner } from './index';
+
+jest.mock('../services', () => ({
+  ...jest.requireActual('../services'),
+  fetchEnterpriseLearnerData: jest.fn().mockResolvedValue(null),
+}));
+const mockAuthenticatedUser = authenticatedUserFactory();
+const mockEnterpriseLearnerData = {
+  enterpriseCustomer: {
+    uuid: 'test-enterprise-customer',
+    slug: 'test-enterprise-slug',
+  },
+  enterpriseCustomerUserRoleAssignments: [],
+  activeEnterpriseCustomer: null,
+  activeEnterpriseCustomerUserRoleAssignments: [],
+  allLinkedEnterpriseCustomerUsers: [],
+  enterpriseFeatures: {},
+  staffEnterpriseCustomer: null,
+};
+
+describe('useEnterpriseCustomer', () => {
+  const Wrapper = ({ children }) => (
+    <QueryClientProvider client={queryClient()}>
+      <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+        {children}
+      </AppContext.Provider>
+    </QueryClientProvider>
+  );
+  beforeEach(() => {
+    jest.clearAllMocks();
+    fetchEnterpriseLearnerData.mockResolvedValue(mockEnterpriseLearnerData);
+  });
+  it('should return nested hook value correctly', async () => {
```

Comment:
IMHO, it's a bit odd to have a test case for `useEnterpriseLearner` when this test suite is about `useEnterpriseCustomer`. Presumably any rendering of `useEnterpriseLearner` is already covered in its own test suite. Pretty sure you can get by with only rendering/testing `useEnterpriseCustomer` directly within this test suite.